compiler/scenario-service/server/src/main/scala/com/digitalasset/daml/lf/ScenarioServiceMain.scala

```diff
@@ -91,14 +132,19 @@ class ScenarioService(implicit
   override def runScenario(
       req: RunScenarioRequest,
       respObs: StreamObserver[RunScenarioResponse],
-  ): Unit =
-    run(
-      req,
-      respObs,
-      { case (ctx, pkgId, name) =>
-        Future.successful(ctx.interpretScenario(pkgId, name))
-      },
-    )
+  ): Unit = {
+    if (enableScenarios) {
+      run(
+        req,
+        respObs,
+        { case (ctx, pkgId, name) =>
+          Future.successful(ctx.interpretScenario(pkgId, name))
+        },
+      )
+    } else {
+      log("Rejected scenario gRPC request.")
```

Comment:
I think this doesn’t quite work. You still need to send an error response to the client. Something like
```
        respObs.onError(new UnsupportedOperationException("Scenarios are disabled"))
```

Reply:
aha! will fix, thank you